core/src/main/java/org/elasticsearch/bootstrap/JavaVersion.java

```diff
@@ -33,6 +33,13 @@
     }
 
     private JavaVersion(List<Integer> version) {
+        if (version.size() >= 2
+                && version.get(0).intValue() == 1
+                && version.get(1).intValue() == 8) {
+            // for Java 8 there is ambiguity since both 1.8 and 8 are supported,
+            // so we rewrite the former to the latter
+            version = new ArrayList<>(version.subList(1, version.size()));
```

Comment:
I had to add it because of https://issues.apache.org/jira/browse/LUCENE-7292. Is there a better way?